backend/manager/modules/vdsbroker/src/main/java/org/ovirt/engine/core/vdsbroker/VdsManager.java

```diff
@@ -294,6 +294,7 @@ public void refreshImpl() {
                 setIsSetNonOperationalExecuted(false);
                 synchronized (this) {
                     refreshCachedVds();
+                    updateCpuTopology();
```

Comment:
we call here the cpuTopology update and within we use `cachedVds`, but based on 2 lines below it may be null.

Reply:
+1